mono/metadata/sgen-thread-pool.c

```diff
@@ -39,6 +39,9 @@ static SgenThreadPoolThreadInitFunc thread_init_func;
 static SgenThreadPoolIdleJobFunc idle_job_func;
 static SgenThreadPoolContinueIdleJobFunc continue_idle_job_func;
 
+static int threadpool_shutdown;
+static int thread_finished;
```

Comment:
Both of these are boolean, so they need to be `gboolean`.  They also need to be `volatile`.